packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -560,6 +565,42 @@ function dc(
     return c(sel, Ctor, data, children);
 }
 
+/**
+ * [dc] - create a dynamic component via `<lwc:component lwc:is={Ctor}>`
+ */
+function ndc(
+    Ctor: LightningElementConstructor | null | undefined,
+    data: VElementData,
+    children: VNodes = EmptyArray
+): VCustomElement | null {
+    if (process.env.NODE_ENV !== 'production') {
+        assert.isTrue(isObject(data), `dc() 2nd argument data must be an object.`);
+        assert.isTrue(
+            arguments.length === 3 || isArray(children),
+            `dc() 3rd argument data must be an array.`
+        );
+    }
+    // null or undefined values should produce a null value in the VNodes
+    if (isNull(Ctor) || isUndefined(Ctor)) {
+        return null;
+    }
+
+    if (!isComponentConstructor(Ctor)) {
+        throw new Error(
+            `Invalid constructor ${toString(Ctor)} is not a LightningElement constructor.`
+        );
+    }
+
+    const sel = getComponentRegisteredName(Ctor);
+    if (isUndefined(sel)) {
+        throw new Error(
+            `Invalid LWC constructor ${toString(Ctor)} does not have a registered name`
+        );
+    }
+
+    return c(sel, Ctor, data, children, true);
+}
+
```

Comment:
After[ part 1 ](https://github.com/salesforce/lwc/pull/3337#discussion_r1100982031)is merged, this function will be the `dc` function.  This is only listed here as dc to ensure the unit tests pass but is not invoked in this PR.